### PR TITLE
update dataviz demo pages

### PR DESCRIPTION
**Pull request by theiliad**
- Changes `basic` charts to `simple`
- Changes `advanced` charts to `complex`
- Updates demos
- Adds the `10 more demos` cards to each demo group

TODO:
- [x] Separate out the stacked area demos in `@carbon/charts`

### CODE REVIEW

**Comment by theiliad** [comment]
Closes #2175

**Comment by mjabbink** [comment]
@theiliad This is awesome. So much better for our users!

— Noticed Storybook icon missing in first resource card
Screen Shot 2021-04-12 at 8 07 50 PM


— Some spacing issue. Image and text too close. @jeanservaas will know spec but the Gatsby should not being doing this I don’t think.
Screen Shot 2021-04-12 at 8 07 50 PM

**Comment by mjabbink** [comment]
Should Sparkline be shown on an axis grid?

**Comment by mjabbink** [comment]
Will Bullet chart be in Complex or Simple? I think we decided in Simple no?

**Comment by mjabbink** [comment]
@theiliad Can we change these situation in storybook site so we use either a em dash between these situations: 

  (time series—dense data) 

or do it like

  (time series/dense data)
  
  @jeanservaas what do you think? I guess if we can keep hyphens but remove space but I think the above two options are better.

**Review by joshblack — APPROVED** [review]

**Comment by theiliad** [comment]
> Should Sparkline be shown on an axis grid?

no sparkline doesn't show up on axes, at least that's the use case it was designed for.

Here's what it looks like with all of the elements visible
image

we can add this demo later if we need to

**Comment by theiliad** [comment]
> Will Bullet chart be in Complex or Simple? I think we decided in Simple no?

sure we can do that, once it's out

**Comment by theiliad** [comment]
we're good to go here, please squash & merge

@jeanservaas we're going to need thumbnails soon for bullet & circle pack

**Comment by theiliad** [comment]
> @jeanservaas we're going to need thumbnails soon for bullet & circle pack

I guess we can merge now without those icons bc we actually need icons for combo, lollipop, sparkline, circle pack, boxplot, wordcloud, bullet & floating bar. Will have to be a separate PR in a week or 2

**Comment by mjabbink** [comment]
@theiliad I like the axis elements in place. That’s where and how it would be utilized. Looks strange to me without it. @jeanservaas what are your thought on Sparline not having the axis structure.

**Comment by mjabbink** [comment]
@theiliad I also think you can merge before @jeanservaas get’s the thumbnails. @jeanservaas Do you agree?

**Comment by tw15egan** [comment]
@theiliad can you [sign the DCO](https://github.com/carbon-design-system/carbon-website/pull/2223#issuecomment-818322629)?

File: src/components/data-visualization/ChartDemoGroup.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    demo = demoGroup.demos[0];
+  }
+
+  const DemoComponent = ChartComponents[demo.chartType.vanilla];
+
+  const numberOfRemainingDemos =
+    demoGroup.demos.length > 2 ? demoGroup.demos.length - 1 : 0;
+
+  return (
+    <>
```
Comment (vpicone) [inline src/components/data-visualization/ChartDemoGroup.js:34]: You probably just want to use a proper `div` here so the page's stack styles get applied if you inspect elements on the other page you can kinda see what's happening:

Screen Shot 2021-04-13 at 1 56 18 PM

Modifying the `--space` css custom property either through react style prop or css would be the preferred way to override the stack spacing behavior.

**Comment by theiliad** [comment]
> _**I have read the DCO document and I hereby sign the DCO.**_

I have read the DCO document and I hereby sign the DCO.

**Comment by theiliad** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by theiliad** [comment]
> I have read the DCO document and I hereby sign the DCO.

@tw15egan I don't think this is working

**Comment by tw15egan** [comment]
recheck

**Comment by theiliad** [comment]
Good thanks

**Review by jillianhowarth — APPROVED** [review]
LGTM 🎉

**Comment by mjabbink** [comment]
👍

**Comment by mjabbink** [comment]
Page looks great. Thank you @theiliad! This is a huge improvement for anyone coming to this page.

**Review by vpicone — APPROVED** [review]
rewrites look good

**Pull request merged**